atomicapp/nulecule/main.py

```diff
@@ -252,12 +252,21 @@ def run(self, cli_provider, answers_output, ask,
             self._write_answers(answers_output, runtime_answers,
                                 self.answers_format)
 
-    def stop(self, cli_provider, **kwargs):
+        try:
+            self.nulecule.run(cli_provider, dryrun)
+        except Exception as e:
+            logger.error('Application run error: %s' % e)
+            logger.debug('Nulecule run error: %s' % e, exc_info=True)
+            logger.info('Rolling back changes')
+            self.stop(cli_provider, ignore_errors=True, **kwargs)
```

Comment:
So in this case we still need to error out after the "stop" has been performed. Otherwise the application returns a good exit code and the user might not realize that it didn't fail.